railties/lib/rails/commands/runner/runner_command.rb

```diff
@@ -51,14 +51,24 @@ def perform(code_or_file = nil, *command_argv)
               eval(code_or_file, TOPLEVEL_BINDING, __FILE__, __LINE__)
             end
           rescue SyntaxError, NameError => e
-            error "Please specify a valid ruby command or the path of a script to run."
-            error "Run '#{executable} -h' for help."
-            error ""
-            error e
+            if looks_like_a_file_path?(code_or_file)
+              error "The file #{code_or_file} could not be found, please check and try again."
```

Comment:
What happens if the file exists but contains a syntax error? I think it would enter this branch and be even more confusing.

Reply:
If the rule has errors they will be raised outside or the begin/rescue block here: https://github.com/rails/rails/blob/7088003d1928870aec5a76a5f8bafb55b89db6f4/railties/lib/rails/commands/runner/runner_command.rb#L46

Reply:
🤦 my bad, I misread the nesting.

Looks fine by me then.